ospfd/ospf_interface.c

```diff
@@ -821,14 +821,41 @@ int ospf_if_up(struct ospf_interface *oi)
 	return 1;
 }
 
-int ospf_if_down(struct ospf_interface *oi)
+/* This function will mark routes with next-hops matching the down
+ * OSPF interface as changed. It is used to assure routes that get
+ * removed from the zebra RIB when an interface goes down are
+ * reinstalled if the interface comes back up prior to an intervening
+ * SPF calculation.
+ */
+static void ospf_if_down_mark_routes_changed(struct route_table *table,
+					     struct ospf_interface *oi)
 {
-	struct ospf *ospf;
 	struct route_node *rn;
 	struct ospf_route *or;
 	struct listnode *nh;
 	struct ospf_path *op;
 
+	for (rn = route_top(table); rn; rn = route_next(rn)) {
+		or = rn->info;
+
+		if (!or)
```

Comment:
looks like the style-check doesn't like this - how about `if (or == NULL)` instead?

Reply:
Yeah - this is my second iteration as "git-clang-format --diff" suggested "if (! or)" but then that failed one of the platform builds. I'll try this and squash the commit. The funny thing is there are plenty of other null pointer checks in FRR using "!ptr". 

Thanks,
Mark

Reply:
yes, I know the "pointer as boolean" thing is common; don't know why this particular instance is setting off the CI check.